src/libraries/System.Private.CoreLib/src/System/IO/Stream.cs

```diff
@@ -492,13 +492,12 @@ private void RunReadWriteTaskWhenReady(Task asyncWaiter, ReadWriteTask readWrite
                 {
                     Debug.Assert(t.IsCompletedSuccessfully, "The semaphore wait should always complete successfully.");
                     var rwt = (ReadWriteTask)state!;
-                    Debug.Assert(rwt._stream != null);
```

Comment:
Why did you delete the assert?

Reply:
I thought it was there because of the instance call (it does not seem to be used inside RunReadWriteTask)

Reply:
There are some confusing code paths here as part of optimizations, but basically RunReadWriteTask invokes some code that will end up nulling out _stream, so this assert is helping to validate that this code isn't run a second time (if it was, it'd be null).

Please put back the assert :)

Reply:
Done (with a comment)